File: packages/rune-keyboard/android/RuneKeyboard/src/main/java/dev/rune/keyboard/RuneKeyboard.kt
package dev.rune.keyboard

import android.app.Activity
import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneKeyboard module
 */
object RuneKeyboard {
    private var moduleInstance: RuneKeyboardModule? = null

    /**
     * Initialize the keyboard module with an Activity and RuneRuntime instance
     * Should be called during app initialization before the JS bundle loads
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: RuneRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("RuneKeyboard", "Module already initialized")
            return
        }

        moduleInstance = RuneKeyboardModule(activity, runtime)
        android.util.Log.d("RuneKeyboard", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown)
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance?.onDestroy()
        moduleInstance = null
    }
}

File: packages/rune-keyboard/ios/RuneKeyboard.swift
//
//  RuneKeyboard.swift
//  RuneKeyboard
//
//  Public interface for RuneKeyboard module
//

import Foundation
import RuneKit

@objc(RuneKeyboard)
public class RuneKeyboard: NSObject {
  
  private static var moduleInstance: RuneKeyboardModule?
  
  /// Initialize the keyboard module with a RuneRuntime instance
  /// Should be called during app initialization before the JS bundle loads
  @objc public static func initialize(with runtime: RuneRuntime) {
    guard moduleInstance == nil else {
      print("[RuneKeyboard] Module already initialized")
      return
    }
    
    // Use the static initialize method from RuneKeyboardModule
    moduleInstance = RuneKeyboardModule.initialize(with: runtime)
    print("[RuneKeyboard] Module initialized")
  }
  
  /// Clean up the module (called during app teardown)
  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
